src/app/(landing)/oauth-callback/facebook/page.tsx

```diff
@@ -0,0 +1,113 @@
+"use client"
+
+import { Suspense, useEffect, useState } from "react"
+import { useRouter, useSearchParams } from "next/navigation"
+import { useAuth } from "@/providers/auth-provider"
+import { useTurnkey } from "@turnkey/sdk-react"
+import { Loader } from "lucide-react"
+
+import { Card, CardHeader, CardTitle } from "@/components/ui/card"
+import { Icons } from "@/components/icons"
+
+import { exchangeToken } from "../../../../actions/turnkey"
+import { verifierSegmentToChallenge } from "../../../../lib/facebook-utils"
+
+function FacebookProcessCallback() {
+  const searchParams = useSearchParams()
+
+  const { authIframeClient } = useTurnkey()
+  const { loginWithFacebook } = useAuth()
+
+  const [storedCode, setStoredCode] = useState<string | null>(null)
+  const [storedState, setStoredState] = useState<string | null>(null)
+  const [hasLoggedIn, setHasLoggedIn] = useState(false)
+
+  const router = useRouter()
+
+  const getToken = async () => {
+    const verifier = await verifierSegmentToChallenge(storedState || "")
+    const token = await exchangeToken(storedCode || "", verifier)
+
+    return token
+  }
+
+  // Get token from query string params and store in state when available
+  useEffect(() => {
+    const code = searchParams.get("code")
+    const state = searchParams.get("state")
+    if (!code && !state) {
+      const inheretedError = searchParams.get("error")
+      let error = "Invalid redirect parameters"
+      if (inheretedError) {
+        error = "Facebook failed with error: " + error
+      }
+      router.push(`/?error=${encodeURIComponent(error)}`)
+    }
+    if (code) {
+      setStoredCode(code)
+    }
+    if (state) {
+      setStoredState(state)
+    }
+  }, [searchParams])
```

Comment:
I think we'd want router as a dep here as well
```suggestion
  }, [searchParams, router])
```